JSON API and views

mszynka opened the pull request:
Adds JSON API in controllers
Adds ModelEditors, Create, Edit and Delete views
Adds Logic for actions
Reimplements contollers
Adds menu dropdowns
Closes #6

mszynka committed:
Merge branch 'preview-01' into feature-post-json-api

mszynka committed:
Project API, Display name, ModelEditor, refactoring

-Project JSON API
-Project properties display name
-Project model editor
-Project actions
-Json Helpers refactoring

mszynka committed:
Merge branch 'preview-01' into feature-post-json-api

Conflicts:
	TaskVlopper.Tests/TaskVlopper.Tests.csproj

mszynka committed:
Model serializers

mszynka committed:
ModelEditors, custom actions, ISerializers and more Json magic

mszynka commented:
I'm not so sure about those views. @bojakowsky Should I use ViewModel or just Model?

mszynka committed:
View to ViewModel conversion

mszynka committed:
Validation messages and ViewModel bugfix
Closes #21

bojakowsky committed:
Merge pull request #22 from bojakowsky/feature-validation-message

Fixes ViewModels

bojakowsky commented:
ViewModel as name suggests is for the view, I've just started correcting stuff

mszynka committed:
Bootstrap update, datepicker and css bundle clean

mszynka commented:
Yeah, I've made a mistake and fixed it. Merge #25 to this for feature and cleanup.

bojakowsky committed:
- Restructurizating code
- Added Logic,
- Registered Logic,
- Mappings are now generic
- Added Serialization Classes and interfaces
- Added Logger class
- WORK IS IN PROGRESS. I JUST GO TO SLEEP :U

bojakowsky commented:
I go to sleep, I'll end it up as I wake up. 
Closed #23

mszynka committed:
Merge branch 'feature-post-json-api' into feature-datepicker

mszynka committed:
IProjectRepository void bugfix
Fixes #29

mszynka committed:
Link network and delete fix
Closes #26

bojakowsky commented:
- UserID in database is a string. That's why, i'll correct it
- var ass in assignments was funny
- BaseSerializer... it's not hardcoded, it's just how mapper MUST work. one type enum =! all other enums, but however I had no time to check if typeof(enum) will work
- Sorry for building eror, it was really later hours, and miss typed types, lol.

bojakowsky committed:
Merge pull request #31 from bojakowsky/feature-link-network

Link network

mszynka commented:
I don't mind build error :wink: And I didn't read the commit message. Sorry :imp: 

Anyway this `typeof(enum)` would be nice. Or could we implement like a `enum interface` and then `typeof` would be explicit. We should be able to. It's a :cl: class, isn't it?

bojakowsky committed:
Merge pull request #25 from bojakowsky/feature-datepicker

Feature datepicker - all OK

bojakowsky committed:
Finnished cleaning up Project stuff (logic, repo), fixed ProjectSetup tests,
added methods to ProjectLogic and Repository

bojakowsky committed:
- Added place to hold model instances, forms passing values etc,
- Extremly important change in ProjectRepository AsNoTracking()
- Ended writing tests for Project

bojakowsky committed:
Restructurized Task stuff, implementation of repo, logic methods needs to be done, tests to be done

bojakowsky committed:
Structurized Worklog, implementation gotta be done, tests gotta be done

bojakowsky committed:
Meeting controllers clean up, logic, repo to be implemented, tests to be done

bojakowsky committed:
fixed logger path

bojakowsky committed:
Merge branch 'feature-post-json-api' of https://github.com/bojakowsky/TaskVlopper.git

bojakowsky commented:
If you're okay with the changes merge PR and close it.
I left many things unimplemented, I'll open another issue for that tomorrow.
Mainly i structurized code, made it more objective.
Contact me via facebook about the eventual problems, I'm having to much spam on gmail to handle that.
fixed #6 #30

mszynka commented on TaskVlopper.Base/Logic/IMeetingLogic.cs:
```diff
@@ -0,0 +1,20 @@
+    {
+        IEnumerable<TaskVlopper.Base.Model.Meeting> GetAllMeetingsForCurrentUser(string userId);
+        IEnumerable<TaskVlopper.Base.Model.Meeting> GetAllMeetingsForCurrentUserAndProject(string userId, int projectId);
+        IEnumerable<TaskVlopper.Base.Model.Meeting> GetAllMeetingsForCurrentUserAndProjectAndTask(string userId, int projectId, int taskId);
```
Why 3 methods for the same task? Couldn't we use like `GetAllMeetingsForCurrent(string userId = null, int projectId = null...)`? And then you would just use `GetAllMeetingsForCurrent(userId: mystring)`

bojakowsky replied:
cause it's not intuitive, after I'll implement them u'll see what i'm talking about

mszynka replied:
It's so intuitive that now we don't have Project.GetAll :smiley: So we need new interface in :cl: ProjectController

mszynka commented on TaskVlopper.Base/Logic/IWorklogLogic.cs:
```diff
@@ -0,0 +1,19 @@
+    public interface IWorklogLogic
+    {
+        IEnumerable<TaskVlopper.Base.Model.Worklog> GetAllWorklogForGivenProjectAndUser(int projectId, string userId);
+        IEnumerable<TaskVlopper.Base.Model.Worklog> GetAllWorklogForGivenProjectAndTaskAndUser(int projectId, int taskId, string userId);
```
The same question why no overloaded method and 2 complicated names instead?

mszynka commented:
Approved
